bukkit/src/main/java/com/denizenscript/depenizen/bukkit/extensions/worldguard/WorldGuardPlayerExtension.java

```diff
@@ -0,0 +1,112 @@
+package com.denizenscript.depenizen.bukkit.extensions.worldguard;
+
+import com.denizenscript.depenizen.bukkit.extensions.dObjectExtension;
+import com.denizenscript.depenizen.bukkit.support.Support;
+import com.denizenscript.depenizen.bukkit.support.plugins.WorldGuardSupport;
+import com.sk89q.worldguard.bukkit.WorldGuardPlugin;
+import com.sk89q.worldguard.protection.ApplicableRegionSet;
+import com.sk89q.worldguard.protection.flags.DefaultFlag;
+import com.sk89q.worldguard.protection.flags.Flag;
+import com.sk89q.worldguard.protection.flags.StateFlag;
+
+import net.aufdemrand.denizen.objects.dLocation;
+import net.aufdemrand.denizen.objects.dPlayer;
+import net.aufdemrand.denizencore.objects.Element;
+import net.aufdemrand.denizencore.objects.dObject;
+import net.aufdemrand.denizencore.tags.Attribute;
+import net.aufdemrand.denizencore.utilities.debugging.dB;
+
+import org.bukkit.entity.Player;
+
+public class WorldGuardPlayerExtension extends dObjectExtension {
+
+    public static boolean describes(dObject object) {
+        return object instanceof dPlayer
+                && ((dPlayer) object).isOnline();
+    }
+
+    public static WorldGuardPlayerExtension getFrom(dObject object) {
+        if (!describes(object)) {
+            return null;
+        }
+        return new WorldGuardPlayerExtension((dPlayer) object);
+    }
+
+    private WorldGuardPlayerExtension(dPlayer player) {
+        this.player = player.getPlayerEntity();
+        this.wgp = Support.getPlugin(WorldGuardSupport.class);
+    }
+
+    Player player = null;
+    WorldGuardPlugin wgp = null;
+
+    // All regions at the players location
+    private ApplicableRegionSet getApplicableRegions() {
+        return wgp.getRegionManager(player.getWorld()).getApplicableRegions(player.getLocation());
+    }
+
+    // The original flag-reference without fuzziness
+    private StateFlag getStateFlag(String string) {
+        for (Flag<?> flag : DefaultFlag.getFlags()) {
+            if (flag.getName().equalsIgnoreCase(string)) {
+                return (StateFlag) flag;

```

Comment:
Are there other kinds of flags (that aren't "StateFlag"s)? If so, what happens when you try to test one here?

Reply:
Yeah, there are StringFlags i.e. for messages - which lead to an exception when queries by a user.
http://old.mcmonkey.org/paste/47831

This should be prevented by the instanceof check now, as you suggested.
It leads then to the debug message.
http://old.mcmonkey.org/paste/47832

I also renamed "string". If these is more to format, let me know.